lib/internal/streams/state.js

```diff
@@ -2,19 +2,19 @@
 
 const { ERR_INVALID_OPT_VALUE } = require('internal/errors').codes;
 
+function highWaterMarkFrom(options, isDuplex, duplexKey) {
+  return options.highWaterMark != null ? options.highWaterMark :
+    isDuplex ? options[duplexKey] : null;
+}
+
 function getHighWaterMark(state, options, duplexKey, isDuplex) {
-  let hwm = options.highWaterMark;
+  const hwm = highWaterMarkFrom(options, isDuplex, duplexKey);
   if (hwm != null) {
-    if (typeof hwm !== 'number' || !(hwm >= 0))
-      throw new ERR_INVALID_OPT_VALUE('highWaterMark', hwm);
-    return Math.floor(hwm);
-  } else if (isDuplex) {
-    hwm = options[duplexKey];
-    if (hwm != null) {
-      if (typeof hwm !== 'number' || !(hwm >= 0))
-        throw new ERR_INVALID_OPT_VALUE(duplexKey, hwm);
-      return Math.floor(hwm);
+    if (typeof hwm !== 'number' || !(hwm >= 0)) {
+      const name = isDuplex ? duplexKey : 'highWaterMark';
+      throw new ERR_INVALID_OPT_VALUE(`options.${name}`, hwm);
```

Comment:
The only reason for making this change was to be consistent with https://github.com/nodejs/node/pull/20284 but I'm happy to remove this.

Reply:
The error constructors are different though. This one is more specific since it's not `ERR_INVALID_ARG_TYPE` but rather `ERR_INVALID_OPT_VALUE`. Meaning it already communicates that it's within the options object.